components/mpas-ocean/src/shared/mpas_ocn_tracer_surface_restoring.F

```diff
@@ -414,11 +422,14 @@ subroutine ocn_get_surfaceSalinityData( streamManager, &
 
         endif
 
+        indexForReproSum(1) = 1
+        indexForReproSum(2) = 0
```

Comment:
This is probably just do to the construction of the repro sum, but I'm a bit confused by the two length array of indexForReproSum.  It seems position 1 is always just 1 and position 2 is the total number of entries in the sum.  Is that right?  What is position 1 of this index array used for?  It never appears to be modified and then just get used in the globalSum call.

Reply:
@vanroekel  The reproducible sums in MPAS allow sums over full arrays and over restricted address ranges in each dimension. For full generality, the range must be specified in min/max pairs for each array dimension. In this case, the array is getting packed in advance so the min is always 1 but the max can vary. But both need to be provided for the interface.

Reply:
Thanks @philipwjones for the explanation. It didn't make sense to me either.

Reply:
@vanroekel this has always been part of the algorithm (essentially the "if deltaS .ne. 0.0" part) to account for possible cases where we explicitly don't want to apply restoring to some cells (eg, under land ice maybe?).  then we don't actually want to include every cell in these sums.  does that make sense?

Reply:
```suggestion
        indexForReproSum(2) = 0
      ! For reproducible sums like mpas_global_sum_mod(), the range must be specified in min/max pairs 
      ! for each array dimension. In this case, the array is getting packed in advance so the min is always 1 
      ! but the max can vary. Both need to be provided for the interface.
```

Reply:
Thanks @philipwjones  for the help understanding this.  I'd like to include the comment from @alicebarthel in the PR for clarity.  If it is too long for a comment, perhaps we can think of a shorter version.